### ISSUE TITLE
Raw Black/White Point Crop Feature

### ISSUE THREAD

**Issue by RyanHarvieStark** [darktable-org/darktable#17140]
I changed cameras recently and was having problems with Panasonic RW2 RAW files not being cropped enough on import. Artifacts often appearing at edges. Changing the darkablerc to:

plugins/darkroom/rawprepare/allow_editing_crop=TRUE

allows setting the crop in the raw black/white point module to the sensor border settings that are within the exif. This avoids any issues by going right to or very near the sensor edge.

There should be an option in preferences to enable crop settings within the raw black/white point module since Rawspeed can mess this up.

**Comment by ralfbrown** [comment]
In most cases, the raw crop should not be messed with....  That's presumably why it can only be enabled by editing the config file, i.e. by advanced users (or under their direction).

**Comment by RyanHarvieStark** [comment]
In Ansel the crop is enabled by default. It doesn't need to be enabled but there should be a preference for crop because RAW files with Rawspeed can have problems in the way they crop unlike Libraw which I was able to test in Ansel because there is a preference option for that. If I wasn't able to use this crop, Darktable would be no good or at least too much trouble to use.

**Comment by MStraeten** [comment]
if you need that option, you can enable this once via darktablerc - except of some edge cases users never needs that so it doesn’t make sense to expose those options by default.

**Comment by RyanHarvieStark** [comment]
OK but it means I'll have to change that file whenever I upgrade Darktable. Not a major issue I suppose but I can see that sometimes you might want to change that. You might want the crop to be the same as the jpg camera output so it's 6000x4000 for example.

There might even be a scenario where you would want to use Libraw instead of Rawspeed.

**Comment by RyanHarvieStark** [comment]
Also, this may effect all Panasonic cameras with RW2 since the proper crop is obviously important although many people might not notice the issue.

**Comment by victoryforce** [comment]
> OK but it means I'll have to change that file whenever I upgrade Darktable.

No, absolutely not. Updating the program does not affect the user settings file in any way.

**Comment by RyanHarvieStark** [comment]
Yeah. I suppose it won't change that file since it's in the home directory. I'm thinking of other stuff I run on here that does get configuration files replaced.

**Comment by jenshannoschwalm** [comment]
There is slightly more involved with changing the raw clips, a pr is pending for this.

**Comment by jenshannoschwalm** [comment]
In current master we now have the rawprepare-cropped dimensions available so it would be safe to make this option available.

Nevertheless i would be opposed to making this option visible via preferences or per default.
dt should always give proper defaults via the raw-loaders, the rawprepare cropper is a dev-tool for checking camera sensors.

If dt forks handle this differently is absolutely no argument for me. Ansel for example does **not** offer some darktable algorithms like raw chromatic aberration correction, the new highlights corrections or embedded lens correction. Those would be prone to non-optimal results. I am pretty sure most users would not know this and would not expect such issues. There are raw files that could feed NaN or other invalid data into the pipe ...

So for me this would definitely not be an "enhancement" but "absolutely not planned" :-)

**Comment by marc-fouquet** [comment]
> [...] except of some edge cases users never needs that so it doesn’t make sense to expose those options by default [...]

> [...] I am pretty sure most users would not know this and would not expect such issues. There are raw files that could feed NaN or other invalid data into the pipe [...]

I am relatively new to the project and really interested in the bigger picture regarding UI decisions.

So may I ask for the difference between this setting and existing settings that are only needed in rare cases and potentially dangerous?

I would argue that all options in the "raw black/white point" module are only needed in rare cases.

There is the black level correction in the exposure module, which has a warning tooltip and also a warning in the manual. However, it is prominently positioned, literally right below the most important slider in all of darktable. It should usually not be needed and has the potential to send bad data into the pipeline.

Has there ever been a discussion about moving such options into an "advanced options" area of their respective modules, so they are not immediately visible, but require an extra click to appear? This could be less confusing for new users and also less visual clutter for regular users.

**Comment by jenshannoschwalm** [comment]
> I would argue that all options in the "raw black/white point" module are only needed in rare cases.

Right. Although there are cameras around with non-optimal white/black point settings. At least some sonys and some elder nikons with 12bit sensors ...

> So may I ask for the difference between this setting and existing settings that are only needed in rare cases and potentially dangerous?

In other modules - like the mentioned black level correction - there might be problems with changed settings but those problems will be a) visible at once while changing and b) will not change the way other modules work. 

The data provided in totally uncropped sensor might contain strange "things" depending on vendor decision. We simply don't know what data is there but assume everything is raw data (uint16_t) so at least there will be artifacts, not even always visible as lens correction might step in giving strange data likely at the corners.

Lens correction will still work (somehow) but if the raw-cropping change is more than a few photosites there **will** be inferior data after lens module and we certainly don't want that. Most likely in CA correction BTW.

So i personally would avoid to UI expose any parameters that might confront the user with unexpected results.
And - we would have to make sure the user always can understand what happens.

Last not least, having the user confronted with non-optimal/incorrect raw crops for a camera, we can help by mentioning the hidden option but also increase pressure to report & properly fix the specific camera settings for dt and rawspeed project. :-)  

@RyanHarvieStark
1. would you be able to provide such an image ?
2. Is this from an "unsupported" camera? If so, please upload at pixls and create an issue in https://github.com/darktable-org/rawspeed
